libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1650,27 +1676,16 @@ impl NodeAPI for Greenlight {
         &self,
         lsp_info: &LspInformation,
     ) -> NodeResult<(Vec<RouteHint>, bool)> {
-        let mut hints: Vec<RouteHint> = vec![];
         let mut node_client = self.get_node_client().await?;
-        // Get the peer channels
-        let peer_channels = node_client
-            .list_peer_channels(cln::ListpeerchannelsRequest::default())
-            .await?
-            .into_inner();
 
-        let mut has_public_channel = false;
-        let open_peer_channels: HashMap<Vec<u8>, cln::ListpeerchannelsChannels> = peer_channels
-            .channels
-            .into_iter()
-            .filter(|c| {
-                let is_private = c.private.unwrap_or_default();
-                has_public_channel |= !is_private;
-                is_private
-                    && c.state == Some(cln::ChannelState::ChanneldNormal as i32)
-                    && c.peer_id.is_some()
-            })
-            .map(|c| (c.peer_id.clone().unwrap(), c))
-            .collect();
+        let open_peer_channels = self.get_open_peer_channels_pb().await?;
+        let has_public_channel = open_peer_channels.values().any(|c| {
```

Comment:
This seems wrong as the get_open_peer_channels_pb filters public channels.

Reply:
Fixed: 47ae85e972215a87b02f238cbe82846f2a19991d